parastell/invessel_build.py

```diff
@@ -536,6 +517,221 @@ def generate_rib(self):
+
+    Optional attributes:
+        plasma_mat_tag (str): alternate DAGMC material tag to use for plasma.
+            If none is supplied, 'plasma' will be used (defaults to None).
+        sol_mat_tag (str): alternate DAGMC material tag to use for scrape-off
+            layer. If none is supplied, 'sol' will be used (defaults to None).
+    """
+
+    def __init__(
+        self,
+        toroidal_angles,
+        poloidal_angles,
+        wall_s,
+        radial_build,
+        logger=None,
+        **kwargs
+    ):
+        
+        self.logger = logger
+        self.toroidal_angles = toroidal_angles
+        self.poloidal_angles = poloidal_angles
+        self.wall_s = wall_s
+        self.radial_build = radial_build
+
+        self.plasma_mat_tag = 'Vacuum'
+        self.sol_mat_tag = 'Vacuum'
+
+        allowed_kwargs = ['plasma_mat_tag', 'sol_mat_tag']
+        set_kwarg_attrs(
+            self,
+            kwargs,
+            allowed_kwargs,
+            self._logger
+        )
+
+        self._logger.info(
+            'Constructing radial build...'
+        )
+
+    @property
+    def toroidal_angles(self):
+        return self._toroidal_angles
+    
+    @toroidal_angles.setter
+    def toroidal_angles(self, angle_list):
+        self._toroidal_angles = angle_list
+        if self._toroidal_angles[0] != 0.0:
+            e = ValueError(
+                'The first entry in toroidal_angles must be 0.0.'
+            )
+            self._logger.error(e.args[0])
+            raise e
+        if self._toroidal_angles[-1] > 360.0:
+            e = ValueError(
+                'Toroidal extent cannot exceed 360.0 degrees.'
+            )
+            self._logger.error(e.args[0])
+            raise e
+
+    @property
+    def poloidal_angles(self):
+        return self._poloidal_angles
+    
+    @poloidal_angles.setter
+    def poloidal_angles(self, angle_list):
+        self._poloidal_angles = angle_list
+        if self._poloidal_angles[-1] - self._poloidal_angles[0] > 360.0:
+            e = AssertionError(
+                'Poloidal extent must span exactly 360.0 degrees.'
+            )
+            self._logger.error(e.args[0])
+            raise e
+
+    @property
+    def wall_s(self):
+        return self._wall_s
+    
+    @wall_s.setter
+    def wall_s(self, s):
+        self._wall_s = s
+        if self._wall_s < 1.0:
+            e = ValueError(
+                'wall_s must be greater than or equal to 1.0.'
+            )
+            self._logger.error(e.args[0])
+            raise e
+        
+        if hasattr(self, 'radial_build'):
+            self.radial_build = self.radial_build
+    
+    @property
+    def radial_build(self):
+        return self._radial_build
+    
+    @radial_build.setter
+    def radial_build(self, build_dict):
+        self._radial_build = build_dict
+        
+        if self._wall_s == 1.0 and 'sol' in self._radial_build:
+            del self.radial_build['sol']
```

Comment:
Doesn't this prevent someone from defining a SOL with varying thicknesses?

Reply:
Yes, you're right. I'll need to come up with some way to delete `'sol'` from `radial_build` only under the right conditions

Reply:
Or maybe non-existent?